util/reggen/reg_top.sv.tpl

```diff
@@ -731,8 +731,8 @@ ${finst_gen(sr, field, finst_name, fsig_name, fidx)}
   assign addrmiss = (reg_re || reg_we) ? ~|addr_hit : 1'b0 ;
 % if racl_support:
   // A valid address hit but failed the RACL check
-  assign racl_error_o = tl_i.a_valid &
-                        (|addr_hit) & ~(|(addr_hit & (racl_addr_hit_read | racl_addr_hit_write)));
+  assign racl_error_o = ~((reg_re & (|racl_addr_hit_read)) |
+                          (reg_we & (|racl_addr_hit_write)));
```

Comment:
This is a nice clean-up. I think I can see why it's now correct, and it's also simpler to read. Nice!

But please could you leave a note in the commit message that explains what's going on? Commit messages aren't really optional.

Reply:
I changed the error message